app/src/Components/Base/Content.js

```diff
@@ -8,14 +8,51 @@ import DataSource from './../DataSource/DataSource';
 import NotFoundComponent from '../Error/NotFoundComponent'
 import Login from '../Login/Login';
 
+const PrivateRoute = ({ component: Component, ...rest }) => (
+    <Route {...rest} render={(props) => (
+        isAuthentificated(rest)
+            ? <Component {...props} />
+            : <Redirect to={{
+            pathname: '/login',
+            state: { from: props.location }
+        }} />
+    )} />
+);
+
+function isAuthentificated(props){
+    let cookieValue = getCookieValue('token');
+    if(cookieValue !== null && cookieValue.length > 5){
+        let token = parseJwt(cookieValue);
+
+        for(let a = 0; a < props.permissions.length; a++) {
+            if(!token.role.includes(props.permissions[a])) {
+                return false;
+            }
+        }
+        return true;
+    }
+    return false;
+}
+
+function parseJwt (token) {
+    let base64Url = token.split('.')[1];
+    let base64 = base64Url.replace('-', '+').replace('_', '/');
```

Comment:
Maybe we should use some kind of library to parse the JWT tokens? What do you think @PascalHonegger @vilkazz @alexisrolland ?

Reply:
Delayed the decision on that for now.